### PR TITLE
Fix/app version upgrade

### PR DESCRIPTION
**Pull request by gokulavasan**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java [hunk 1]
```diff
@@ -628,6 +630,45 @@ public boolean apply(RunRecordMeta runRecordMeta) {
...
+  private static MDSKey appendDefaultVersion(MDSKey oldKey) {
+    MDSKey.Splitter splitter = oldKey.split();
+    int keyParts = 0;
+    try {
+      for (int i = 0; i < 4; i++) {
+        splitter.skipString();
+        keyParts++;
+      }
+    } catch (BufferUnderflowException ex) {
+      // expected
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java]: when does this happen?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java [hunk 2]
```diff
@@ -821,6 +821,15 @@ void clear() throws Exception {
     truncate(dsFramework.getAdmin(WORKFLOW_STATS_INSTANCE_ID.toId(), null));
   }
 
+  public void upgradeAppVersion() throws Exception {
+    txExecute(transactional, new TxRunnable() {
+      @Override
+      public void run(DatasetContext context) throws Exception {
+        getAppMetadataStore(context).upgradeVersionKeys();
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java:828]: safe to do all updates in a single transaction?

**Review by bdmogal — COMMENTED** [review]
Couple questions. But lgtm otherwise

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java (hunk 2 above)
Comment (gokulavasan) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java:828]: I think so.

**Comment by gokulavasan** [comment]
Merging. Upgrade tested on a cluster.

**Pull request merged**